include/stdlib.h

```diff
@@ -12,105 +12,785 @@ extern "C" {
+ * On failure, returns a null pointer.
+ * */
+void *malloc(size_t size);
 
 /**
-* Free previously allocated memory
-*/
-void free(void* ptr);
+ * @brief Deallocates allocated memory space.
+ *
+ * Deallocates the space previously allocated by @see malloc, @see calloc, @see realloc.
+ * If ptr is a null pointer, the function does nothing.
+ *
+ * The behavior is undefined if the value of ptr does not equal a value returned earlier
+ * by @see malloc, @see calloc, @see realloc.
+ *
+ * The behavior is undefined if the memory area referred to by ptr has already been deallocated,
+ * that is, free() or realloc has already been called with ptr as the argument and no calls to
+ * malloc, calloc or realloc resulted in a pointer equal to ptr afterwards.
+ *
+ * The behavior is undefined if after free returns, an access is made through the pointer ptr
+ * (unless another allocation function happened to result in a pointer value equal to ptr)
+ *
+ * @param ptr pointer to the memory to deallocate
+ * @return none
+ * */
+void free(void *ptr);
 
-void* calloc(size_t num, size_t size);
+/**
+ * @brief Allocates memory for an array of given number objects of size
+ *        and initializes all bytes in the allocated storage to zero.
+ *
+ * Allocates memory for an array of num objects of size size and initializes
+ * all bytes in the allocated storage to zero.
+ *
+ * @param num number of objects
+ * @param size size of the array
+ * @return (None)
+ * */
+void *calloc(size_t num, size_t size);
 
 /**
-* realloc(ptr, 0) is a special case that can be handled in two ways. See:
-*	 http://pubs.opengroup.org/onlinepubs/9699919799/functions/realloc.html
-* realloc(ptr, 0) returns NULL.
-*/
-void* realloc(void* ptr, size_t size);
+ * @brief Reallocates the given area of memory.
+ *
+ * Reallocates the given area of memory. It must be previously allocated by @see malloc,
+ * @see calloc or @see realloc and not yet freed with a call to @see free or @see realloc.
+ * Otherwise, the results are undefined.
+ *
+ * The reallocation is done by either:
+ *          1) expanding or contracting the existing area pointed to by ptr, if possible.
+ *             The contents of the area remain unchanged up to the lesser of the new and old sizes.
+ *             If the area is expanded, the contents of the new part of the array are undefined.
+ *          2) allocating a new memory block of size size bytes, copying memory area with size
+ *             equal the lesser of the new and the old sizes, and freeing the old block.
+ *
+ * If there is not enough memory, the old memory block is not freed and null pointer is returned.
+ * If ptr is NULL, the behavior is the same as calling malloc(size).
+ *
+ * If size is zero, the behavior is implementation defined (null pointer may be returned
+ * (in which case the old memory block may or may not be freed), or some non-null pointer may
+ * be returned that may not be used to access storage).
```

Comment:
For the special case, we have a definite behavior (I selected): realloc(ptr, 0) returns NULL.